### ISSUE TITLE
Support WaveRNN inference with different sampling mode

### ISSUE THREAD

**Issue by yangarbiter** [pytorch/audio#1721]
Currently, the WaveRNN pipeline supports MOLLoss but there are no pretrained weight trained with that loss yet. In the future, we want to support pretrained weight trained with MOLLoss and the corresponding sampling mode for it.

Right now, the inference function for WaveRNN only supports multinomial sampling. (#1704)

One implementation for the sampling model for MOLLoss is [here](https://github.com/fatchord/WaveRNN/blob/master/models/fatchord_version.py#L226-L229).
